scripts/perf/lib.sh

```diff
@@ -0,0 +1,292 @@
+# strategies.
+
+set -e
+set -x
+
+# When run in testing-farm we'll use the TMT_PLAN_DATA to store all artifacts
+# from the functions below.
+# (For TMT_PLAN_DATA see https://tmt.readthedocs.io/en/stable/overview.html#step-variables)
+RESULT_DIR=${TMT_PLAN_DATA:-/tmp}
+YYYYMMDD=${YYYYMMDD:-$(date +%Y%m%d)}
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+# How many times do we want to run a test and later take the average mean?
+NUM_TEST_RUNS=${NUM_TEST_RUNS:-2}
+
+# Construct chroot if needed
+CHROOT=${COPR_CHROOT:-}
+if [[ -z "$CHROOT" ]]; then
+    os=$(cat /etc/os-release | grep -oP '^ID=["]?\K[^"]+')
+    product_version=$(cat /etc/os-release | grep -oP "REDHAT_BUGZILLA_PRODUCT_VERSION=\K.*")
+    arch=$(uname -m)
+    CHROOT=${os}-${product_version}-${arch}
+fi
+
+# Helper function to configure, build and test the llvm-test-suite in a
+# directory with the given `NAME` and dump the test results in a file called
+# `<RESULT_DIR>/<NAME>.json` for later comparison.
+function _configure_build_test {
+    local NAME=$1
+    local N_BUILD_JOBS=$2
+    local ITH_BUILD=$3
+
+    local BUILD_DIR=builds/$NAME
+
+    mkdir -pv $BUILD_DIR
+    pushd $BUILD_DIR
+
+    # See also https://llvm.org/docs/TestSuiteGuide.html#common-configuration-options
+    local cmake_args=""
+
+    # Recommended setting for compile-time benchmarks
+    #
+    # TODO(kwk): Should we test for performance criterias other than compile
+    # time, we might need to adjust this.
+    cmake_args="$cmake_args -DTEST_SUITE_SUBDIRS=CTMark"
+
+    # For PGO performance comparison we expect differences in the range of 10%
+    # and more. Therefore we don't necessarily need perf. On containers we need
+    # to turn it off.
+    if [[ -n "$container" ]]; then
+        cmake_args="$cmake_args -DTEST_SUITE_USE_PERF=OFF"
+    else
+        cmake_args="$cmake_args -DTEST_SUITE_USE_PERF=ON"
+    fi
+
+    # We want to measure the run-time of the compiler and therefore don't have
+    # to "run" the benchmarks. We just need to compile them.
+    cmake_args="$cmake_args -DTEST_SUITE_RUN_BENCHMARKS=OFF"
+
+    # Collect internal LLVM statistics. Appends -save-stats=obj when invoking
+    # the compiler and makes the lit runner collect and merge the statistic
+    # files.
+    cmake_args="$cmake_args -DTEST_SUITE_COLLECT_STATS=ON"
+
+    # Some programs are unsuitable for performance measurements. Setting the
+    # TEST_SUITE_BENCHMARKING_ONLY CMake option to ON will disable them.
+    cmake_args="$cmake_args -DTEST_SUITE_BENCHMARKING_ONLY=ON"
+
+    # Configure the test suite
+    cmake \
+        -GNinja \
+        -DCMAKE_C_COMPILER=/usr/bin/clang \
+        -DCMAKE_CXX_COMPILER=/usr/bin/clang++ \
+        $cmake_args \
+        -C/usr/share/llvm-test-suite/cmake/caches/O3.cmake \
+        /usr/share/llvm-test-suite
+
+    # Build the test-suite with one job at a time for a fair comparison.
```

Comment:
Out-of-date comment.

Reply:
Fixed in https://github.com/fedora-llvm-team/llvm-snapshots/pull/1275/commits/b6c63f4876875a43e4f63aced0189b81b54be937